rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java

```diff
@@ -667,6 +681,87 @@ protected Integer commitFederation(boolean dryRun, Keccak256 hash, BridgeEventLo
         return FederationChangeResponseCode.SUCCESSFUL.getCode();
     }
 
+    /**
+     * Commits the currently pending federation to be the proposed one.
+     * That is, the proposed federation is set to be the currently pending federation,
+     * and the pending federation is wiped out.
+     * @param dryRun whether to just do a dry run
+     * @param hash the pending federation's hash. This is checked the execution block's pending federation hash for equality.
+     * @return 1 upon success, -1 if there was no pending federation, -2 if the pending federation was incomplete,
+     * -3 if the given hash doesn't match the current pending federation's hash.
+     */
+    private Integer commitFederationPostRSKIP419(boolean dryRun, Keccak256 hash, BridgeEventLogger eventLogger) {
+        PendingFederation currentPendingFederation = provider.getPendingFederation();
+
+        if (currentPendingFederation == null) {
+            logger.warn("[commitFederation] Pending federation does not exist.");
+            return FederationChangeResponseCode.PENDING_FEDERATION_NON_EXISTENT.getCode();
+        }
+
+        if (!currentPendingFederation.isComplete()) {
+            logger.warn("[commitFederation] Pending federation has {} members, so it does not meet the minimum required.", currentPendingFederation.getMembers().size());
+            return FederationChangeResponseCode.INSUFFICIENT_MEMBERS.getCode();
+        }
+
+        if (!hash.equals(currentPendingFederation.getHash())) {
+            logger.warn("[commitFederation] Provided hash {} does not match pending federation hash {}.", hash, currentPendingFederation.getHash());
+            return FederationChangeResponseCode.PENDING_FEDERATION_MISMATCHED_HASH.getCode();
+        }
+
+        if (dryRun) {
+            logger.info("[commitFederation] DryRun execution successful.");
+            return FederationChangeResponseCode.SUCCESSFUL.getCode();
+        }
+
+        // Network parameters for the proposed federation are taken from the bridge constants.
+        // Creation time is the block's timestamp.
```

Comment:
nit, but I don't think this comment is adding any value. No big deal though. We can discuss it with the rest of the team.